rust/src/cliwrap/yumdnf.rs

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{anyhow, Result};
+use indoc::indoc;
+use std::os::unix::process::CommandExt;
+use std::process::Command;
+use structopt::StructOpt;
+
+const UNHANDLED: &str = indoc::indoc! {r#"
+rpm-ostree: Unknown arguments passed to yum/dnf compatibility wrapper.
+
+Please see `yum --help` for currently accepted commands via this
+wrapper, and `rpm-ostree --help` for more information about.the rpm-ostree
+project.
+"#};
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "yumdnf-rpmostree",
+    about = "Compatibility wrapper implementing subset of yum/dnf CLI"
+)]
+#[structopt(rename_all = "kebab-case")]
+/// Main options struct
+enum Opt {
+    /// Start an upgrade of the operating system
+    Upgrade,
+    /// Start an upgrade of the operating system
+    Update,
+    /// Display information about system state
+    Status,
+    /// Perform a search of packages.
+    Search {
+        /// Search terms
+        terms: Vec<String>,
+    },
+    /// Will return an error suggesting other approaches.
+    Install {
+        /// Set of packages to install
+        packages: Vec<String>,
+    },
+    /// Layer RPM packages as operating system extensions
+    InstallExtension {
+        /// Set of packages to install
+        packages: Vec<String>,
+    },
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum RunDisposition {
+    HelpOrVersionDisplayed,
+    ExecRpmOstree(Vec<String>),
+    UseSomethingElse(&'static str),
+    Unhandled,
+}
+
+fn disposition(argv: &[&str]) -> Result<RunDisposition> {
+    let opt = match Opt::from_iter_safe(std::iter::once(&"yum").chain(argv.iter())) {
+        Ok(v) => v,
+        Err(e)
+            if e.kind == clap::ErrorKind::VersionDisplayed
+                || e.kind == clap::ErrorKind::HelpDisplayed =>
+        {
+            return Ok(RunDisposition::HelpOrVersionDisplayed)
+        }
+        Err(_) => {
+            return Ok(RunDisposition::Unhandled);
+        }
+    };
+
+    let disp = match opt {
+        Opt::Upgrade | Opt::Update => RunDisposition::ExecRpmOstree(vec!["upgrade".into()]),
+        Opt::Status => RunDisposition::ExecRpmOstree(vec!["status".into()]),
+        Opt::Install { packages: _ } => {
+            // TODO analyze packages to find e.g. `gcc` (not ok, use `toolbox`) versus `libvirt` (ok)
+            RunDisposition::UseSomethingElse(indoc! {r##"
+This system is rpm-ostree based.  Before installing packages to
+the host root filesystem, consider other options:
+
+ - `toolbox`: For command-line development and debugging tools in a privileged container
+ - `podman`: General purpose containers (run a webserver, e.g.)
+ - `flatpak`: For desktop (GUI) applications
+ - `yum install-extension`: The same thing as `rpm-ostree install --apply-live`.  Treat packages as OS extensions.
```

Comment:
Hmm, feels like enhancing the yum/dnf CLI with new commands is one step further and may be more confusing than helpful.

Reply:
```suggestion
 - `rpm-ostree install`: For packages which need to be directly on the host. Treat these as "OS extensions". Add `--apply-live` to immediately start using the layered packages.
```

?

Reply:
OK, let's try to take the details of stuff like this to a hackmd doc that turns into a GH issue or so.  We could (and honestly, will) debate the details of this for a long time.  My goal here is just to add a strawman starting point - remember this is all off by default.